packages/website/src/docs/extending-cadl/emitters-basics.md

```diff
@@ -0,0 +1,155 @@
+---
+eleventyNavigation:
+  key: Writing Emitters
+  parent: Extending Cadl
+  order: 2
+---
+
+# Writing emitters
+
+Cadl emitters are libraries that use various Cadl compiler APIs to reflect on the cadl compilation and produce generated artifacts. The cadl standard library includes an emitter for OpenAPI version 3.0, but odds are good you will want to emit Cadl to another output format. In fact, one of Cadl's main benefits is how easy it is to use Cadl as a source of truth for all data shapes, and the ease of writing an emitter is a big part of that story.
+
+## Getting started
+
+Cadl emitters are a special kind of Cadl library and so have the same getting started instructions. Follow [these steps](#todo) to initialize a cadl library.
+
+## $onEmit
+
+A Cadl emitter exports a function named `$onEmit`. It receives two arguments:
+
+- _program_: The current `Program` instance being compiled
+- _options_: Custom configuration options selected for this emitter
+
+For example, the following will write a text file to the output directory:
+
+```typescript
+import { Program } from "@cadl-lang/compiler";
+import Path from "path";
+
+export async function $onEmit(program: Program) {
+  const outputPath = Path.join(program.compilerOptions.outputPath!, "hello.txt");
+  await program.host.writeFile(outputPath, "hello world!");
+}
+```
+
+You can now compile a Cadl program passing your library name to --emit, or add it to your `cadl-project.yaml`.
+
+### Custom configuration options
```

Comment:
Like in the other PR I think it would be good to give a best practice on when an emitter should define a decorator vs an emitter option

Reply:
Added some rationale, take a look.

Reply:
I don't see it, did you forget to push?